### PR TITLE
FEATURE: TNT-1792 Increase PivotTable bucket size limit

### PR DESCRIPTION
**Pull request by NestorEncinas**
Will increase max bucket size for Pivot Table only on Panther - 

- Measures: 100
- Rows: 50
- Columns: 20 (no changes)

JIRA: TNT-1792

---

Supported PR commands:

| Command                                                 | Description                                                |
| ------------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                            | Re-run standard checks                                     |
| `extended check sonar`                                  | SonarQube tests                                            |
| `extended test - backstop`                              | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                  |                                                            |
| `extended test - tiger-cypress - isolated <testName>`   | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`     | Create a new recording for isolated Tiger tests.           |
| `extended test - tiger-cypress - integrated <testName>` | Run integrated tests against live backend                  |
| **E2E Cypress tests commands - BEAR**                   |                                                            |
| `extended test - cypress - isolated <testName>`         | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`           | Create a new recording for isolated Bear tests.            |
| `extended test - cypress - integrated <testName>`       | Run integrated tests against live backend                  |
| **Compatibility matrix test commands - TIGER Backend**  |                                                            |
| `extended test - matrix-test <AIO_version>`             | Run integrated tests against AIO versions.                 |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

`<AIO_version>` in commands is used to start test with multiple AIO instances - each instance in triggered by one jenkins build

-   To run with `master` and `stable`, execute command `extended test - matrix-test master,stable` or `extended test - matrix-test latest`
-   To run with specific version,ex: `2.3.0` and `2.3.1`, execute command `extended test - matrix-test 2.3.0,2.3.1`
-   In case `<AIO_version>` is empty, read versions from file `compTigerVersions.txt` of this repo

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `extended test - tiger-cypress - integrated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16252/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17753/ : FAILURE in 9m 04s

File: libs/sdk-ui-ext/src/internal/translations/en-US.json [hunk 1]
```diff
@@ -576,6 +576,21 @@
         "limit": 0,
         "translate": false
     },
+    "dashboard.bucket.category_columns_warning._measure": {
+        "value": "To add columns, total items from measures or rows should not be higher than 20",
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui-ext/src/internal/translations/en-US.json:580]: can we use placeholder here too, smth like {oldLimit} ?

File: libs/sdk-ui-ext/src/internal/utils/uiConfigHelpers/pivotTableUiConfigHelper.ts [hunk 2]
```diff
@@ -17,23 +19,67 @@ import {
...
+            let warningMessageId;
+            let warningMessageValues = {};
+
+            if (bucket.localIdentifier === BucketNames.COLUMNS) {
+                warningMessageId = messages.columns.id;
+            } else {
+                warningMessageId = messages.measuresAttributes.id;
+
+                if (bucket.localIdentifier === BucketNames.MEASURES) {
+                    warningMessageValues = { limit: "80" };
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui-ext/src/internal/utils/uiConfigHelpers/pivotTableUiConfigHelper.ts:59]: evaluate it as newLimit - oldLimit? To keep it auto updated when one of limits will change

File: libs/sdk-ui-ext/src/internal/utils/uiConfigHelpers/pivotTableUiConfigHelper.ts [hunk 3]
```diff
@@ -17,23 +19,67 @@ import {
...
+                } else {
+                    warningMessageValues = { limit: "30" };
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui-ext/src/internal/utils/uiConfigHelpers/pivotTableUiConfigHelper.ts:61]: same as above

File: libs/sdk-ui-ext/src/internal/utils/bucketRules.ts [hunk 4]
```diff
@@ -76,6 +76,17 @@ export function hasNoAttribute(buckets: IBucketOfFun[]): boolean {
     return getItemsCount(buckets, BucketNames.ATTRIBUTE) === 0;
 }
 
+export function hasNoMoreThan20ItemsOnMeasuresOrRows(buckets: IBucketOfFun[]): boolean {
+    return (
+        getItemsCount(buckets, BucketNames.ATTRIBUTE) <= 20 &&
```
Comment (ivan-nejezchleb) [inline libs/sdk-ui-ext/src/internal/utils/bucketRules.ts:81]: use constants for old limits

**Review by ivan-nejezchleb — CHANGES_REQUESTED** [review]

File: libs/sdk-ui-ext/src/internal/translations/en-US.json (hunk 1 above)
Comment (NestorEncinas) [reply libs/sdk-ui-ext/src/internal/translations/en-US.json]: Yes, added it.

File: libs/sdk-ui-ext/src/internal/utils/uiConfigHelpers/pivotTableUiConfigHelper.ts (hunk 2 above)
Comment (NestorEncinas) [reply libs/sdk-ui-ext/src/internal/utils/uiConfigHelpers/pivotTableUiConfigHelper.ts]: Done.

File: libs/sdk-ui-ext/src/internal/utils/bucketRules.ts (hunk 4 above)
Comment (NestorEncinas) [reply libs/sdk-ui-ext/src/internal/utils/bucketRules.ts]: Done.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16258/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17760/ : SUCCESS in 12m 12s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16259/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17762/ : SUCCESS in 12m 13s

**Review by ivan-nejezchleb — APPROVED** [review]

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR4228

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/3337/ : SUCCESS in 13m 22s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17764/ : SUCCESS in 12m 44s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-pipeline/945/ : SUCCESS in 10m 11s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/4499/ : SUCCESS in 14s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/1217/ : FAILURE in 1m 30s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul-pipeline/851/ : SUCCESS in 8m 56s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-compatibility-pipeline/494/ : SUCCESS in 7m 24s

**Pull request merged**